pkg/storageincentives/redistribution/redistribution.go

```diff
@@ -23,9 +23,9 @@ type Contract interface {
 	ReserveSalt(context.Context) ([]byte, error)
 	IsPlaying(context.Context, uint8) (bool, error)
 	IsWinner(context.Context) (bool, error)
-	Claim(context.Context) error
-	Commit(context.Context, []byte, *big.Int) error
-	Reveal(context.Context, uint8, []byte, []byte) error
+	Claim(context.Context) (*big.Int, error)
+	Commit(context.Context, []byte, *big.Int) (*big.Int, error)
+	Reveal(context.Context, uint8, []byte, []byte) (*big.Int, error)
```

Comment:
Would it make sense if `Claim`, `Commit` and `Reveal` method return transaction hash `common.Hash` to the caller, instead of fee? 

If so, these would be advantages:
- This way caller would be able to have more information (should it need) and not just fee.
- Caller would be able to collect all transaction hashes and calculate fees in separate goroutne. This would reduce total execution time of  of `Claim`, `Commit` and `Reveal` methods, because with current change they have one more operation to execute (fetch transaction).
- Error handling for case when transaction could not be fetched could be better handled (instead of swallowing it here)

Reply:
this is also a good idea, probably fits the nature of the change better, as fee and reward calculations should not affect the participation in the game

Reply:
Sounds good.